### ISSUE TITLE
Navigation Component: Add `ul` Automatically Where Needed

### ISSUE THREAD

**Issue by Copons** [WordPress/gutenberg#25248]
`NavigationMenu` can contains multiple lists or `NavigationGroup`s, so it doesn't wrap its children with an `ul`.
On the other hand, `NavigationItem` is a `li`, so it needs to live inside an `ul`.
See the [story](https://github.com/WordPress/gutenberg/blob/668a710b3eb9c942b21eb5913fbe304b15ea71dd/packages/components/src/navigation/stories/index.js) for an example.

Possible solutions:

- By traversing the `NavigationMenu` children, we can determine if it needs an `ul` and automatically add it, instead of offloading the burden to the consumer.
Depends on #25246 exploration.

- Make `NavigationMenu` always render as an `ul`, and `NavigationGroup` as a `<li><ul /></li>`.
In other words, if a menu contains only one list, it will be a simple lists; otherwise it will be a _list of lists_ (whereas right now is a "series" of lists).

See https://github.com/WordPress/gutenberg/pull/25057#discussion_r485983226
